app/main.py

```
@@ -4309,4 +4309,6 @@ async def get_newsletter():
     return res
 
 if __name__ == "__main__":
-    uvicorn.run(app, host="0.0.0.0", port=8000)
+    api_port = int(os.environ.get('API_PORT'))
+    uvicorn_base_url = os.environ.get('UVICORN_BASE_URL')
```

one thing i usually do for parameters i expect to receive as an input is to collect all the values, then ensure they were all passed. this way we can get the code to fail early and spit out a message about what it expected to be passed but wasnt. i think this is the case for API keys and maybe this URL

something like: 
```
base_url = os.environ.get('UVICORN_BASE_URL', None)
api_key = os.environ.get('API_KEY', None)

if base_url is none or api_key is None:
  print("one or more parameters missing. expecting the following variables: API_KEY, UVICORN_BASE_URL ")
  sys.exit(1)
```

---

https://github.com/ekinolik/stocknear-backend/pull/4/files#r1940592004 is probably a better way to do this!

---

updated docker-compose